src/main/java/seedu/address/ui/RoomTaskListPanel.java

```diff
@@ -24,65 +25,59 @@
     @FXML
     private ScrollPane roomScrollPane;
     @FXML
-    private VBox roomBox;
+    private ListView<Task> taskListView;
 
     /**
      * Creates a {@code RoomTaskListPanel} with the given {@code ObservableList}.
      */
-    public RoomTaskListPanel(ObservableList<Room> roomTaskList) {
+    public RoomTaskListPanel(ObservableList<Task> roomTaskList) {
         super(FXML);
-        populatePanel(roomTaskList);
         updateDetailsIfChanged(roomTaskList);
-    }
-
-    /**
-     * Populates the {@code RoomTaskListPanel} with the {@code Room} containing at least one {@code Task}.
-     *
-     * @param roomTaskList The list of rooms containing tasks with which to populate the panel.
-     */
-    private void populatePanel(ObservableList<Room> roomTaskList) {
-        for (Room room : roomTaskList) {
-            roomBox.getChildren().add(new RoomTaskDetailsPanel(room).getRoot());
-        }
+        taskListView.setItems(roomTaskList);
+        taskListView.setCellFactory(listView -> new TaskListViewCell());
 
-        logger.info("RoomTaskListPanel has been populated.");
```

Comment:
i think it would still be good to log the process though

Reply:
Oh the whole populatePanel function is deleted haha so the logger is gone.

Reply:
yup, what i'm thinking about is that it would be good to add a logger to keep track of the changes in the ui

Reply:
Roger that sir.